cylc/flow/task_job_mgr.py

```diff
@@ -1229,7 +1231,12 @@ def _prep_submit_task_job_error(self, workflow, itask, action, exc):
             'submit_num': itask.submit_num,
             'try_num': itask.get_try_num(),
         })
+
         # create a DB entry for the submit-failed job
+        if itask.platform:
+            platform_name = itask.platform['name']
+        else:
+            platform_name = itask.tdef.rtconfig['platform']
```

Comment:
In what case was `itask.platform` not set?

Logically the platform must have been set before the job was submission could fail? Unless maybe platform selection errored?

Reply:
When the submit task job error is a caused by a failure to select a platform - see line 1132

Reply:
The platform is set for me using the same example as in the previous comment and this diff:

```diff
 diff --git a/cylc/flow/task_events_mgr.py b/cylc/flow/task_events_mgr.py
index ea2c7c1be..3b177cb50 100644
--- a/cylc/flow/task_events_mgr.py
+++ b/cylc/flow/task_events_mgr.py
@@ -1207,6 +1207,7 @@ class TaskEventsManager():
 
         # Register newly submit-failed job with the database and datastore.
         job_d = itask.tokens.duplicate(job=str(itask.submit_num)).relative_id
+        print('#', itask.platform)
         self._insert_task_job(itask, event_time, self.JOB_SUBMIT_FAIL_FLAG)
         self.data_store_mgr.delta_job_state(job_d, TASK_STATUS_SUBMIT_FAILED)
 
diff --git a/cylc/flow/task_job_mgr.py b/cylc/flow/task_job_mgr.py
index da995bac8..1f000531d 100644
--- a/cylc/flow/task_job_mgr.py
+++ b/cylc/flow/task_job_mgr.py
@@ -1229,6 +1229,7 @@ class TaskJobManager:
             'submit_num': itask.submit_num,
             'try_num': itask.get_try_num(),
         })
+        print('%', itask.platform)
         # create a DB entry for the submit-failed job
         self.workflow_db_mgr.put_insert_task_jobs(
             itask,
```